index.js

```diff
@@ -180,7 +188,16 @@ function fastifyOauth2 (fastify, options, next) {
     revokeAllTokenCallbacked(token, params, callback)
   }
 
-  const oauth2 = new AuthorizationCode(credentials)
+  const oauth2 = new AuthorizationCode({
+    ...credentials,
+    http: {
```

Comment:
Cant we have an implementation were we dont have a ternary operation and instantiate a new Object for every call?

Reply:
@Uzlopak I reordered it now, but maybe it is this special bit you are against?

```js
...(userAgent ? { 'User-Agent': userAgent } : {}),
```

I could replace it with a straight up assignment if we're okay with it being set to `undefined` when `options.userAgent` is `false`:

```js
'User-Agent': userAgent,
```

Would that be better?

Reply:
@Uzlopak I did this change now, resolving